sale_order_line_qty_delivered_updater/models/sale_order_line.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.multi
+    @api.depends(
+        "qty_invoiced",
+        "qty_delivered_method",
+        "qty_delivered_manual",
+        "analytic_line_ids.so_line",
+        "analytic_line_ids.unit_amount",
+        "analytic_line_ids.product_uom_id",
+    )
+    def _compute_qty_delivered(self):
+        res = super()._compute_qty_delivered()
+        for line in self:
+            if line.qty_invoiced < 0:
+                line.qty_delivered = line.qty_invoiced
+        return res
```

Comment:
This is not enough.  If we are to implement something, we need to check whether the product is the "Amount Adjustment" product.  Therefore we need to have a boolean field in the product which should be only visible for the service type.